contrib/firebaseio/src/main/java/contrib/JacksonCoder.java

```diff
@@ -0,0 +1,152 @@
+
+import com.google.cloud.dataflow.sdk.coders.Coder;
+import com.google.cloud.dataflow.sdk.coders.CoderException;
+import com.google.cloud.dataflow.sdk.coders.StandardCoder;
+import com.google.cloud.dataflow.sdk.util.CloudObject;
+import com.google.cloud.dataflow.sdk.values.TypeDescriptor;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectMapper.DefaultTyping;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * TODO: Insert description here. (generated by elibixby)
+ */
+public class JacksonCoder<T> extends StandardCoder<T> {
+
+  private static final long serialVersionUID = -754345287170755870L;
+
+  protected static final ObjectMapper MAPPER = new ObjectMapper()
+      .enableDefaultTyping(DefaultTyping.NON_FINAL)
+      .enable(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS)
+      .enable(DeserializationFeature.WRAP_EXCEPTIONS)
+      .enable(SerializationFeature.WRAP_EXCEPTIONS)
+      .disable(SerializationFeature.FAIL_ON_UNWRAPPED_TYPE_IDENTIFIERS)
+      .enable(SerializationFeature.WRITE_NULL_MAP_VALUES);
+
+  private final ObjectWriter writer;
+  private final ObjectReader reader;
+
+  protected final Class<T> type;
+
+  public static <K> JacksonCoder<K> of(TypeDescriptor<K> type) {
+    @SuppressWarnings("unchecked")
+    Class<K> clazz = (Class<K>) type.getRawType();
+    return of(clazz);
+  }
+
+  public static <K> JacksonCoder<K> of(Class<K> clazz){
+    return new JacksonCoder<K>(clazz);
+  }
+
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  @JsonCreator
+  public static JacksonCoder<?> of(
+      @JsonProperty("type") String classType) throws ClassNotFoundException {
+    return new JacksonCoder(Class.forName(classType));
+  }
+
+  public JacksonCoder(Class<T> clazz){
+    this.type = clazz;
+    writer = MAPPER.writer();
+    reader = MAPPER.reader(type);
+  }
+
+  /**
+   * @see Coder#encode(java.lang.Object, OutputStream, Coder.Context)
+   **/
+  @Override
+  public void encode(T value, OutputStream outStream,
+      com.google.cloud.dataflow.sdk.coders.Coder.Context context)
+          throws CoderException, IOException {
+    writer.writeValue(outStream, value);
+  }
+
+  /**
+   * @see Coder#decode(InputStream, Coder.Context)
+   **/
+  @Override
+  public T decode(InputStream inStream, com.google.cloud.dataflow.sdk.coders.Coder.Context context)
+      throws CoderException, IOException {
+    return reader.readValue(inStream);
+  }
+
+  /**
+   * @see Coder#getCoderArguments()
+   **/
```

Comment:
Usually you do not need to include these javadocs. By default, if no comments are specified, Javadoc will just include the javadoc on the super function. You usually only need to put anything here if you want to clarify your implementation, e.g.,

```java
/**
 * For my special function, always throws an exception.
 *
 * {@inheritDoc}  // optional, if the super doc is useful.
 */